woocommerce-gateway-stripe.php

```diff
@@ -474,7 +474,7 @@ protected function toggle_upe( $settings, $old_settings ) {
 					return $settings;
 				}
 
-				if ( 'no' === $old_settings[ WC_Stripe_Feature_Flags::UPE_CHECKOUT_FEATURE_ATTRIBUTE_NAME ] && 'yes' === $settings[ WC_Stripe_Feature_Flags::UPE_CHECKOUT_FEATURE_ATTRIBUTE_NAME ] ) {
+				if ( 'yes' === $settings[ WC_Stripe_Feature_Flags::UPE_CHECKOUT_FEATURE_ATTRIBUTE_NAME ] ) {
```

Comment:
If you disable UPE from the new settings then `$old_settings[ WC_Stripe_Feature_Flags::UPE_CHECKOUT_FEATURE_ATTRIBUTE_NAME ]` is set as `disabled` ([from here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/66ecaa434d556c0e47cf9e1e44a04006a9ebac8d/includes/admin/class-wc-stripe-rest-upe-flag-toggle-controller.php#L76)). For this reason when enabling UPE from the new settings after disabling at least once, this condition was resulting in false and this path was not executed. As a result, enabling the corresponding UPE methods as well as disabling the legacy ones was not working in the new settings.

Just checking if the `$settings[ WC_Stripe_Feature_Flags::UPE_CHECKOUT_FEATURE_ATTRIBUTE_NAME ]` is `yes` should be enough here to decide to execute this if block. Because in the previous `if` block it is already checked if the old and current data are the same and it is returned from there. We can safely assume that when the current value is `true` the old value is either `no` or `disabled`. So did not check the old value explicitly here.